bpf/lib/bpf_linux_binprm.h

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Tetragon */
+
+#ifndef __BPF_LINUX_BINPRM_
+#define __BPF_LINUX_BINPRM_
+
+#define MAX_STRING 1024
```

Comment:
Is there a reason to define this vs using a predefinition from limits.h?

Reply:
Yeah, good question. Here I'm following the convention currently being used in other places in the [codebase](https://github.com/dwindsor/tetragon/blob/9d93c9911b6155f2432f67853bba7a5da19eb21b/bpf/process/types/basic.h#L185C10-L185C10). I'll add a comment here to reference the original definition.

Any values we get from `limits.h` be OS-dependent, so I'm not sure that would work well here. We need this limit to be consistent across all platforms.

We could use `PATH_MAX` here, but that would be 4096 bytes, which is larger than the current `MAX_STRING`, and would not be consistent across OSes.

Reply:
What is the length of the filename in the eBPF struct itself?  Because that is really what this should be matched to in this case, right?
If string maximum is not equivalent to filepath name length, could and attacker bypass the system by generating a file that exceeds that string length?

I think this presents a serious problem in a security solution.
Note too that PATH_MAX is an API limit, I believe you can create filenames that exceed that length too.  I think we need a strong comment around all of this and the issue that an attacker can create a file name that exceeds this limit.  

What is the limit of the filename in the eBPF struct itself and does it do shortening when needed? (short path names).

Reply:
> If string maximum is not equivalent to filepath name length, could and attacker bypass the system by generating a file that exceeds that string length?

Yes, the filename would be truncated in userspace. I'm not sure this is dire, as other attributes like the (device, inode) tuple, which is part of every file-based event in Tetragon, can be used for disambiguation along with the possibly shortened path. 

> What is the limit of the filename in the eBPF struct itself and does it do shortening when needed? (short path names).

This is a complicated answer. In Tetragon's kernel code, they're [copying the entire path](https://github.com/dwindsor/tetragon/blob/9d93c9911b6155f2432f67853bba7a5da19eb21b/bpf/process/types/basic.h#L413) without checking for any bounds.

The eBPF struct you refer to is auto-generated. We actually only define this struct as a grpc message, and the Tetragon infrastructure takes care of the rest. 

Looking at the generated code, it appears that Tetragon treats the path argument coming from the kernel as an unbounded go string. From api/v1/tetragon/tetragon.pb.go:

```go
type KprobeLinuxBinprm struct {
        state         protoimpl.MessageState
        sizeCache     protoimpl.SizeCache
        unknownFields protoimpl.UnknownFields

        Path string `protobuf:"bytes,1,opt,name=path,proto3" json:"path,omitempty"`
}
...

func (x *KprobeLinuxBinprm) GetPath() string {
        if x != nil {
                return x.Path
        }
        return ""
}
```

Reply:
Probably best to discuss live when we meet.